src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -610,6 +610,23 @@ public class ClassWithRequiredExtensionDataProperty
             public required Dictionary<string, JsonElement>? TestExtensionData { get; set; }
         }
 
+        [Fact]
+        public async Task RequiredCollectionPropertyWithoutGetterThrows()
+        {
+            string json = """{"Foo":"foo","Bar":"bar"}""";
+            InvalidOperationException exception = await Assert.ThrowsAsync<InvalidOperationException>(
```

Comment:
I don't think this is the correct way to address the problem. A fix should instead try to make the scenario work as expected. If you remove `JsonRequired` from the repro in #104700 then deserialization works as expected and this should too.